### ISSUE TITLE
Event-sending failures not reported

### ISSUE THREAD

**Issue by UnitedMarsupials** [splunk/splunk-library-javalogging#219]
I understand, that calling the `HttpEventCollectorSender.send()` simply adds the event to batch -- to be processed asynchronously -- but some way of reporting submission-failures is needed anyway.

I had the `index`-value miscCapItalized and could not figure out, why the messages aren't getting there. It was only when I employed the network-sniffer, that I saw the server's response:
```json
        HTTP/1.1 400 Bad Request
        Date: Fri, 03 Dec 2021 02:29:16 GMT
        Server: Splunkd
        Content-Type: application/json; charset=UTF-8
        X-Content-Type-Options: nosniff
        Content-Length: 60
        Vary: Authorization
        X-Frame-Options: SAMEORIGIN
        Connection: close

        {"text":"Incorrect index","code":7,"invalid-event-number":1}
```
It needs to be easier to debug, than this...

**Comment by anders4work** [comment]
Are there any news on this?
Looking at the code I did not find any log statements that could help me debug problems with the plugin/configuration.
Can one somehow pipe the errors from the asynchrounous sending to standard out!?

**Comment by anders4work** [comment]
Solved, to my satisfaction at least, using the approach described in the javadoc of the class HttpEventCollectorHandler:

`Splunk http event collector error handler. A user application can utilize HttpEventCollectorErrorHandler in order to detect errors caused by network connection and/or Splunk server. Usage example: HttpEventCollectorErrorHandler.onError(new HttpEventCollectorErrorHandler.ErrorCallback() { public void error(final String data, final Exception ex) { // handle exception } });`

**Comment by bparmar-splunk** [comment]
Hi @anders4work,
We assume your issue has been resolved by making use of HttpEventCollectorErrorHandler class.
Also, we would recommend to update the current logging library to newly released version ([1.11.5](https://github.com/splunk/splunk-library-javalogging/releases/tag/1.11.5)).

Please let us know, if you still face any issues.

**Comment by UnitedMarsupials** [comment]
> A user application can utilize HttpEventCollectorErrorHandler in order to detect errors caused by network connection and/or Splunk server

@bparmar-splunk, the above may be sufficient for an application _programmer_, who knows in advance, that Splunk will be used as the log4j appender. It is not sufficient for an _administrator_, who tries to configure the `log4j2.xml` -- and is unable to introduce his own Java-code into the application to provide a handler.

The whole idea of independent logging implementations (such as log4j) is to separate the programmer from the administrator -- it being necessary to implement own error-handler breaks this paradigm.

As a minimum, some simple handler -- one printing errors to the application's `stderr` -- ought to be included in the code, with the configuration knobs to enable/disable its use available. Perhaps, it should even be turned on by default too.

**Comment by stokpop** [comment]
Hi @UnitedMarsupials and @anders4work, to be able to see errors and timeouts I added an `StandardErrorCallback` that writes to `stderr`, see: https://github.com/splunk/splunk-library-javalogging/issues/214 and https://github.com/splunk/splunk-library-javalogging/pull/215

It can be activated via the config:

`<errorCallback>com.splunk.logging.util.StandardErrorCallback</errorCallback>`

(enable stacktraces via `-Dcom.splunk.logging.util.StandardErrorCallback.enablePrintStackTrace=true`)

Although not clear from the release notes, it should be present in release: [1.11.5](https://github.com/splunk/splunk-library-javalogging/releases/tag/1.11.5)

@bparmar-splunk as a sidenote: should/can the fixes in 1.11.5 be added to the [CHANGELOG.md](https://github.com/splunk/splunk-library-javalogging/blob/main/CHANGELOG.md?) and to the release notes? There are more fixed issues in this release: https://github.com/splunk/splunk-library-javalogging/compare/1.11.4...1.11.5

HTH

**Comment by bparmar-splunk** [comment]
Hi @stokpop,
Thank you for mentioning this point.

We will update release notes & CHANGELOG file which includes issue resolutions & enhancements.
